### PR TITLE
Fixed some warnings in Encamina.Enmarcha.Bot

### PR DESCRIPTION
**Pull request by HugoRamosEs**
- Realized minor refactoring in IMetadataProcessor, IQuestionRequestOptions, QuestionRequestOptions, allowing MetadataOptions to be null to resolve various warnings (CS8602, CS8604).
- Updated ITextTranslationRequest and TextTranslationRequest to specify that CultureInfo can be null, addressing warning CS8603.
- Refactored DialogsProviderBase, IDialogProvider, IDialogTypeProvider, ITendedDialogProvider, INamedDialogProvider, NamedComponentDialogBase, and NamedDialogBase to resolve warnings (CS8602, CS8603, and CS8625).
- Modified SendResponseResult to indicate that ResourceResponse can be null, resolving warning CS8063.
- Performed code cleanup in IServiceCollectionExtensions of Encamina.Enmarcha.Bot.Skills to adjust the positioning of a pragma.
- Updated QuestionAnsweringDialog and QuestionAnsweringDialogServices to address warnings CS8603 and CS8625.
- Modified BotCloudAdapterWithErrorHandlerBase to specify that options can be null, resolving warning CS8603.
- Refactored IServiceCollectionExtensions of Encamina.Enmarcha.Bot to resolve warning CS8603 and ensure the service received cannot be null, in addition to performing minor code cleanup and pragma repositioning.
- Updated LocalizedHeroCardGreetingsOptionsFromTableStorage to resolve warnings CS8603.
- Revised ApplicationInsightsConversationScopedLoggerEventSource to resolve warning CS8603 by making applicationName nullable, and adjusted ApplicationInsightsConversationScopedLogger to resolve warning CS8602 by making Exception nullable.
- Removed the possibility of cognitiveServiceProvider being null in DetectedLanguageTranslatorMiddleware and StartActivityTranslatorMiddleware, and performed minor code cleanup in the descriptions.
- Conducted minor code cleanup in TranslatorUtils and adjusted translations to be nullable to resolve warning CS8603.
- Updated TableStorageResponseProvider to specify that IMemoryCache can be null, resolving warning CS8603.

### CODE REVIEW

**Review by MarioRamosEs — APPROVED** [review]

File: src/Encamina.Enmarcha.Bot/Logging/ApplicationInsigthsConversationScopedLogger.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ public bool IsEnabled(LogLevel logLevel)
     }
 
     /// <inheritdoc />
-    public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+    public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception?, string> formatter)
```
Comment (rliberoff) [inline src/Encamina.Enmarcha.Bot/Logging/ApplicationInsigthsConversationScopedLogger.cs:71]: It does not make sense here for the `formatter` to receive a nullable exception. The `formatter` type should be as it was: `Func<TState, Exception, string>`. If the compiler is giving you an warning here, lets understand it first because the solution might be different and not as simpler as adding an `?` :/

File: src/Encamina.Enmarcha.Bot.Skills.QuestionAnswering/QuestionAnsweringDialog.cs [hunk 2]
```diff
@@ -59,9 +59,9 @@ public QuestionAnsweringDialog(string id, QuestionAnsweringDialogServices questi
     public override string Name => string.IsNullOrWhiteSpace(configurationOptions.DialogName) ? base.Name : configurationOptions.DialogName;
 
     /// <inheritdoc/>
-    public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default)
+    public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object? options = null, CancellationToken cancellationToken = default)
```
Comment (rliberoff) [inline src/Encamina.Enmarcha.Bot.Skills.QuestionAnswering/QuestionAnsweringDialog.cs]: Setting `object?` is somehow breaking the public interface of the `Microsoft.Bot.Builder.Dialogs.Dialog` type.
For now, revert this change and create a new specific issue to address this part of the code.

**Review by rliberoff — CHANGES_REQUESTED** [review]

File: Directory.Build.props [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
   <PropertyGroup>
     <VersionPrefix>8.1.6</VersionPrefix>
```
Comment (MarioRamosEs) [inline Directory.Build.props]: We must update this version

**Review by MarioRamosEs — CHANGES_REQUESTED** [review]

**Review by rliberoff — APPROVED** [review]

**Review by MarioRamosEs — APPROVED** [review]
